src/software/ai/hl/stp/tactic/crease_defender/crease_defender_fsm.cpp

```diff
@@ -30,10 +34,24 @@ std::optional<Point> CreaseDefenderFSM::findBlockThreatPoint(
     return findDefenseAreaIntersection(field, ray, robot_obstacle_inflation_factor);
 }
 
+bool CreaseDefenderFSM::isAnyEnemyInZone(const Update& event, const Stadium& zone)
+{
+    std::vector<Robot> enemy_robots = event.common.world_ptr->enemyTeam().getAllRobots();
+    for (int i = 0; i < static_cast<int>(event.common.world_ptr->enemyTeam().numRobots()); i++) {
+        if (contains(zone, enemy_robots[i].position()))
+        {
+            return true;
+        }
+    }
+    return false;
```

Comment:
nit: try refactoring to use `std::any_of`: https://en.cppreference.com/w/cpp/algorithm/all_any_none_of